ldap_auth_provider.py

```diff
@@ -88,6 +93,20 @@ def check_password(self, user_id, password):
             defer.returnValue(False)
         # user_id is of the form @foo:bar.com
         localpart = user_id.split(":", 1)[0][1:]
+        uid_value = localpart
+        default_givenName = localpart
```

Comment:
please could you add some comments here? what do each of these variables do, and what is the difference between `localpart` and `uid_value` ?